mtags/src/main/scala-2/scala/meta/internal/pc/SignatureHelpProvider.scala

```diff
@@ -89,16 +89,23 @@ class SignatureHelpProvider(val compiler: MetalsGlobal) {
         case tpe => tpe
       }
     }
+    val isUnapplyMethod: Boolean = tree match {
+      case _: UnApply => true
+      // in some case we will get an Apply tree, but in that case position indicates unapply
+      case _ =>
```

Comment:
In what cases we will get Apply tree?

Reply:
No idea, that's what is returned from the compiler.

Reply:
@tgodzik in which case? I replaced this case by returning false and cross/test pass for 3.1.1

Reply:
The test is not enabled at all for Scala 3. I don't remember exactly which of the cases, I think `generic` test case?

Reply:
:facepalm: I missed that it's about scala2 :smiling_face_with_tear:

Reply:
I got back to Scala 2 to figure out what we should expect from the functionality in Scala 3 as well.